### ISSUE TITLE
Consider using simple streaming HTTP connection over Websockets for watches

### ISSUE THREAD

**Issue by ilackarms** [juliusv/prometheus-alert-buffer#23]
What are the advantages of using Websockets over simple HTTP / TCP streaming? 

I ask because websocket protocol is a bit complicated; a client won't be able to simply being a watch with a basic HTTP GET. They'll have to use a websocket client library, most likely (unless they know the websocket protocol and can negotiate the upgrade over basic http).

I think websockets are better suited for bidirectional message passing; however our API does not support client-to-server messages. Since messages are only passed one way, server-to-client, I think streaming HTTP body might be better for our use case. For example, with resource watches, Kubernetes uses HTTP streaming body. 

Thoughts on this?

**Comment by ilackarms** [comment]
Agreed. I can head up a PR on this if you think it would be worth making the change.

**Comment by juliusv** [comment]
@ilackarms I think the change generally makes sense, but in the end it depends on your organizational context whether it's worth it. Probably shouldn't be too much work though. I'm still waiting for a resolution on https://github.com/juliusv/prometheus-alert-buffer/issues/8 to get the repo moved out of my private GitHub account and under Red Hat's ownership, so it's clearer that you can just do whatever you want with it :)

**Comment by ilackarms** [comment]
it's already been forked to https://github.com/openshift/prometheus-alert-buffer :stuck_out_tongue:

**Comment by juliusv** [comment]
@ilackarms Ah perfect. Ok for me to delete this repo here then and open any new issues over there?

**Comment by ilackarms** [comment]
yep i already opened this issue on the fork. maybe you want to add your comment there from above?

also idk what happens to a fork when th eoriginal repo gets deleted

**Comment by juliusv** [comment]
@ilackarms I have no idea either! Should we try and see? I mean, you still have a copy of the code, I assume.

**Comment by ilackarms** [comment]
hahahaha

**Comment by ilackarms** [comment]
go for it

**Comment by juliusv** [comment]
@ilackarms 3... 2... 1...
